lib/lists.nix

```diff
@@ -197,8 +198,9 @@ rec {
     default:
     # Input list
     list:
-    let found = filter pred list;
-    in if found == [] then default else head found;
+    let
+      found = filter pred list;
+    in if found == [ ] then default else head found;
```

Comment:
This get expanded because there is exists newline between `let` and `in`. However, body `in` doesn't get expanded. 

I think the rules should be more relaxed?